sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -101,15 +119,21 @@ object AnalysisContext {
  */
 class Analyzer(
     catalog: SessionCatalog,
+    v2SessionCatalog: TableCatalog,
     conf: SQLConf,
     maxIterations: Int)
   extends RuleExecutor[LogicalPlan] with CheckAnalysis with LookupCatalog {
 
+  // Only for tests.
```

Comment:
I don't link this too much. Do we need to change too much if we need to remove this constructor? It's bad to inject test related stuff in production code

Reply:
There are more than 10 test suites using this constructor. BTW this is an existing constructor, this PR just add a comment.